src/interface/blas1_interface.hpp

```diff
@@ -322,7 +360,43 @@ typename ValueType<container_0_t>::type _dot(executor_t &ex, index_t _N,
   auto gpu_res = make_sycl_iterator_buffer<element_t>(static_cast<index_t>(1));
   blas::internal::_dot(ex, _N, _vx, _incx, _vy, _incy, gpu_res);
   ex.get_policy_handler().copy_to_host(gpu_res, res.data(), 1);
-  return res[0];
+  return res[0]; //TODO Fix this function and add test
```

Comment:
Is this todo something you intended to resolve in this PR?

Reply:
This is an existing function that I think is bugged (but need to add a test to check). I intended to solve this in this Ticket (after doing the other level one operators) but not on this PR. I forgot to remove the TODO when creating the PR.

Reply:
Done